### PR TITLE
Added a slide on span

### LINKED ISSUE #50: C++ 20 : add slides about spans

### PR DESCRIPTION
**Pull request by sponce**
and fixed details on the string_view one

Closes #50

### CODE REVIEW

File: talk/morelanguage/morestl.tex [hunk 1]
```diff
@@ -22,6 +22,39 @@ \subsection{More STL}
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[20]{\texttt{std::span<T>}}
+  \begin{block}{Non owning view of a continuous sequence of items}
+    \begin{itemize}
+    \item Doesn't allocate memory
+    \item Allows to modify items, but not the storage itself
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:30]:
```suggestion
    \item Allows to modify items, but not the container itself
```

File: talk/morelanguage/morestl.tex [hunk 2]
```diff
@@ -22,6 +22,39 @@ \subsection{More STL}
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[20]{\texttt{std::span<T>}}
+  \begin{block}{Non owning view of a continuous sequence of items}
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:27]:
```suggestion
  \begin{block}{Non-owning view of a contiguous sequence of items}
```

File: talk/morelanguage/morestl.tex [hunk 3]
```diff
@@ -22,6 +22,39 @@ \subsection{More STL}
...
+    \begin{itemize}
+    \item Doesn't allocate memory
+    \item Allows to modify items, but not the storage itself
+      \begin{itemize}
+      \item no reallocation, push or pop
+      \end{itemize}
+    \item Provides iterators, \mintinline{cpp}{size}, \mintinline{cpp}{front}/\mintinline{cpp}{back}, \mintinline{cpp}{subspan}, ...
+    \item Standard contiguous containers convert to it automatically
+      \begin{itemize}
+      \item \mintinline{cpp}{std::vector}, \mintinline{cpp}{std::array}, C arrays, ...
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:37]: Should we mention that non-contiguous containers like `std::list`, `std::deque` etc. don't convert?

File: talk/morelanguage/morestl.tex [hunk 4]
```diff
@@ -22,6 +22,39 @@ \subsection{More STL}
...
+      \item should thus be preferred to them when possible
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:38]:
```suggestion
      \item should thus be preferred as function arguments when possible
```

**Review by bernhardmgruber — COMMENTED** [review]
You did not mention spans with static size, like `std::span<int, 64>`, which avoid the storage for the length. Consider mentioning that in one sentence.

**Comment by sponce** [comment]
Added a line on static size, split in 2 slides and thus expanded the examples (including with static size)

File: talk/morelanguage/morestl.tex [hunk 5]
```diff
@@ -22,6 +22,68 @@ \subsection{More STL}
...
+      \item no reallocation, push or pop
+      \end{itemize}
+    \item Provides container like interface
+      \begin{itemize}
+      \item iterators, \mintinline{cpp}{size}, \mintinline{cpp}{front}/\mintinline{cpp}{back}, \mintinline{cpp}{subspan}, ...
+      \end{itemize}
+    \item Standard contiguous containers convert to it automatically
+      \begin{itemize}
+      \item \mintinline{cpp}{std::vector}, \mintinline{cpp}{std::array}, C arrays, ...
+      \item not list or deque as they do not have contiguous storage
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:41]:
```suggestion
      \item not \mintinline{cpp}{std::list} or \mintinline{cpp}{std::deque} as they do not have contiguous storage
```

File: talk/morelanguage/morestl.tex [hunk 6]
```diff
@@ -22,6 +22,68 @@ \subsection{More STL}
...
+      \item \mintinline{cpp}{span} should thus be preferred as function arguments when possible
+      \end{itemize}
+    \item Simply a pair (pointer, size), so cheap to copy
+      \begin{itemize}
+      \item and thus to be passed by value
+      \end{itemize}
+    \item \mintinline{cpp}{span} can also have a \texttt{static extend}
+      \begin{itemize}
+      \item meaning the size is determined at compile time
+      \item and thus only the pointer remains in the \mintinline{cpp}{span}
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:51]:
```suggestion
      \item and thus only a pointer is stored by \mintinline{cpp}{span}
```

**Review by bernhardmgruber — APPROVED** [review]
LGTM, two small suggestions:

File: talk/morelanguage/morestl.tex [hunk 7]
```diff
@@ -22,6 +22,68 @@ \subsection{More STL}
...
+      \item \mintinline{cpp}{span} should thus be preferred as function arguments
+      \end{itemize}
+    \item Simply a pair (pointer, size), so cheap to copy
+      \begin{itemize}
+      \item and thus to be passed by value
+      \end{itemize}
+    \item \mintinline{cpp}{span} can also have a \texttt{static extend}
+      \begin{itemize}
+      \item meaning the size is determined at compile time
+      \item and thus only the a pointer is stored by \mintinline{cpp}{span}
```
Comment (bernhardmgruber) [inline talk/morelanguage/morestl.tex:51]:
```suggestion
      \item and thus only a pointer is stored by \mintinline{cpp}{span}
```

**Review by bernhardmgruber — APPROVED** [review]
Please fix this one quick, and then we are good to go.

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**